packages/flame/test/game/base_game_test.dart

```diff
@@ -200,4 +200,20 @@ void main() {
     game.update(0);
     expect(game.components.isEmpty, equals(true));
   });
+
+  test('clear removes all components', () {
+    final game = MyGame();
+    final components = List.generate(3, (index) => MyComponent());
+
+    game.onResize(size);
+    game.addAll(components);
+
+    // The components are not added to the component list until an update has been performed
+    game.update(0.0);
+    expect(game.components.length, equals(3));
+
+    game.clear();
+    game.update(0.0);
```

Comment:
I would also add an `expect` before the update to ensure that the clear does not clear it directly.

```suggestion
    expect(game.components.length, equals(3));
    game.update(0.0);
```

Reply:
Updates made as per your comments